### PR TITLE
:recycle: (dmk) [DSDK-571]: Refactor ListDeviceSession into ListConnectedDevice

### PR DESCRIPTION
**Pull request by jdabbech-ledger**
### 📝 Description

- Remove `ListDeviceSessionsUseCase`
- Create `ListConnectedDevicesUseCase` and `ListenToConnectedDeviceUseCase`
- Update Sample app
- Update `ConnectedDevice` model to add `sessionId`

### ❓ Context

- **JIRA or GitHub link**: [DSDK-571]

- **Feature**:

### ✅ Checklist

Pull Requests must pass CI checks and undergo code review. Set the PR as Draft if it is not yet ready for review.

- [x] **Covered by automatic tests** 
- [x] **Changeset is provided** 
- [ ] **Impact of the changes:** 
  - _list of the changes_

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.

[DSDK-571]: https://ledgerhq.atlassian.net/browse/DSDK-571?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/device-management-kit/src/api/transport/model/ConnectedDevice.ts [hunk 1]
```diff
@@ -18,8 +21,10 @@ export class ConnectedDevice {
       deviceModel: { id: deviceModelId, productName: deviceName },
       type,
     },
+    sessionId,
   }: ConnectedDeviceConstructorArgs) {
     this.id = id;
+    this.sessionId = sessionId;
```
Comment (mbertin-ledger) [inline packages/device-management-kit/src/api/transport/model/ConnectedDevice.ts:27]: [SHOULD] As id is useless, maybe we should remove id field from `ConnectedDevice` model ?

**Review by mbertin-ledger — CHANGES_REQUESTED** [review]

File: packages/device-management-kit/src/api/transport/model/ConnectedDevice.ts (hunk 1 above)
Comment (jdabbech-ledger) [reply packages/device-management-kit/src/api/transport/model/ConnectedDevice.ts:27]: It's actually used by the hook `useAvailableDevice` of the sample app to match a discovered device to a connected device in order to have the `connected` status of an available device

File: packages/device-management-kit/src/internal/discovery/use-case/ListenToConnectedDeviceUseCase.test.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      { id: fakeSessionId, connectedDevice },
+      () => logger,
+      managerApi,
+    );
+    const observable = new ListenToConnectedDeviceUseCase(
+      sessionService,
+      () => logger,
+    ).execute();
+
+    observable.subscribe({
```
Comment (valpinkman) [inline packages/device-management-kit/src/internal/discovery/use-case/ListenToConnectedDeviceUseCase.test.ts:52]: [ASK] is this observable completing ? if not t means we have leakage in the tests

File: packages/device-management-kit/src/api/transport/model/ConnectedDevice.ts (hunk 1 above)
Comment (mbertin-ledger) [reply packages/device-management-kit/src/api/transport/model/ConnectedDevice.ts:27]: Ok. not a big fan but ok

**Review by mbertin-ledger — APPROVED** [review]

**Pull request merged**